x/gov/msgs.go

```diff
@@ -67,16 +67,16 @@ func (msg MsgSubmitProposal) Get(key interface{}) (value interface{}) {
 // Implements Msg.
 func (msg MsgSubmitProposal) GetSignBytes() []byte {
 	b, err := msgCdc.MarshalJSON(struct {
-		Title          string    `json:"title"`
-		Description    string    `json:"description"`
-		ProposalType   string    `json:"proposal_type"`
-		Proposer       string    `json:"proposer"`
-		InitialDeposit sdk.Coins `json:"deposit"`
+		Title          string         `json:"title"`
+		Description    string         `json:"description"`
+		ProposalType   string         `json:"proposal_type"`
+		Proposer       sdk.AccAddress `json:"proposer"`
+		InitialDeposit sdk.Coins      `json:"deposit"`
 	}{
 		Title:          msg.Title,
 		Description:    msg.Description,
 		ProposalType:   ProposalTypeToString(msg.ProposalType),
```

Comment:
Can we override `MarshalJSON` on `ProposalType` in a similar way?

Doesn't need to block this PR though.

Reply:
Yes, planning on doing that.  Same for some other stuff in governance too.